### PR TITLE
Enhance tika document parsing tests

### PR DESCRIPTION
**Pull request by finnegancarroll**
### Description
Enhance tika document parsing tests by validating output against current version.

### Related Issues
Resolves "Improve the validation on TikaDocTests #12887"

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [ ] ~New functionality has been documented.~
  - [ ] ~New functionality has javadoc added~
- [x] Failing checks are inspected and point to the corresponding known issue(s) (See: [Troubleshooting Failing Builds](../blob/main/CONTRIBUTING.md#troubleshooting-failing-builds))
- [x] Commits are signed per the DCO using --signoff
- [ ] ~Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))~
- [ ] ~Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose)~

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by finnegancarroll** [comment]
Gradle check failing due to unrelated flaky test: #11979

**Comment by finnegancarroll** [comment]
Known flaky test: #10006

**Comment by finnegancarroll** [comment]
Known flaky test: #13476

**Comment by dblock** [comment]
1. Is there any advantage of having those zipped in the repo other than having to unzip them?
2. Keeping a checksum map in code feels a little odd. I don't feel strongly about it, but maybe a .checksum file would be a bit cleaner.

File: plugins/ingest-attachment/src/test/java/org/opensearch/ingest/attachment/TikaDocTests.java [hunk 1]
```diff
@@ -32,54 +32,279 @@
...
-        Path tmp = createTempDir();
-        logger.debug("unzipping all tika sample files");
-        try (DirectoryStream<Path> stream = Files.newDirectoryStream(PathUtils.get(getClass().getResource(TIKA_FILES).toURI()))) {
-            for (Path doc : stream) {
-                String filename = doc.getFileName().toString();
-                TestUtil.unzip(getClass().getResourceAsStream(TIKA_FILES + filename), tmp);
-            }
-        }
+    /** tika 2.9.2 expected output in the form of filename:sha1 key value pairs  */
+    static final Map<String, String> tika292Checksums = Map.ofEntries(
```
Comment (reta) [inline plugins/ingest-attachment/src/test/java/org/opensearch/ingest/attachment/TikaDocTests.java:60]: This looks really like complications and duplication:  we have folder with the files, why do we need to repeat its content as a map? As @dblock mentioned, adding complementary `.sha256` file would help (if needed)

**Comment by finnegancarroll** [comment]
Hi @dblock and @reta thanks for taking a look! I've moved the checksum map to a separate file.

> 1. Is there any advantage of having those zipped in the repo other than having to unzip them?
I believe the intent here is to hide from the linter. testEXCEL.xls for example is not UTF-8 and fails the precommit "forbiddenPatterns" task.

File: plugins/ingest-attachment/src/test/java/org/opensearch/ingest/attachment/TikaDocTests.java [hunk 2]
```diff
@@ -32,54 +32,68 @@
...
-        }
-
-        try (DirectoryStream<Path> stream = Files.newDirectoryStream(tmp)) {
-            for (Path doc : stream) {
-                logger.debug("parsing: {}", doc);
-                assertParseable(doc);
-            }
-        }
+    @Before
+    public void setLocale() {
```
Comment (reta) [inline plugins/ingest-attachment/src/test/java/org/opensearch/ingest/attachment/TikaDocTests.java:63]: We run tests with randomized locales, please remove default local changes unless there are reasons to do that

Comment (finnegancarroll) [reply plugins/ingest-attachment/src/test/java/org/opensearch/ingest/attachment/TikaDocTests.java:63]: Tika uses the locale to help inform it's parsing in some cases. In particular testEXCEL.xls and testEXCEL.xlsx will produce different outputs depending on the locale.

**Comment by finnegancarroll** [comment]
Known flaky test: #13600

File: plugins/ingest-attachment/src/test/java/org/opensearch/ingest/attachment/TikaDocTests.java (hunk 2 above)
Comment (reta) [reply plugins/ingest-attachment/src/test/java/org/opensearch/ingest/attachment/TikaDocTests.java:63]: Our users will be using essentially any locale, we cannot use ENGLISH only, moreover as you said, locale alters the results and we may need to account for that.

Comment (finnegancarroll) [reply plugins/ingest-attachment/src/test/java/org/opensearch/ingest/attachment/TikaDocTests.java:63]: Would it be appropriate to remove only these two files which have locale dependent parsing? Or alternatively we could validate their output across all possible locales. How would you recommend I handle these locale dependent files while satisfying the initial issue?

Comment (reta) [reply plugins/ingest-attachment/src/test/java/org/opensearch/ingest/attachment/TikaDocTests.java:63]: > Would it be appropriate to remove only these two files which have locale dependent parsing?

That would basically means no checksum check, right? May be we could do this:
 - have a special case of empty checksum, so these files would be parsed but not verified
 - or, make checksum locale dependent, in this case if locale is different from what checksum supports, files would be parsed but not verified

Comment (finnegancarroll) [reply plugins/ingest-attachment/src/test/java/org/opensearch/ingest/attachment/TikaDocTests.java]: Thanks for the input. Removed strict checksum verification of locale dependent sample files.

**Comment by reta** [comment]
@finnegancarroll we sadly have pretty flaky test suite now, fe this combination fails for me:

```
./gradlew ':plugins:ingest-attachment:test' --tests "org.opensearch.ingest.attachment.TikaDocTests.testParseSamples" -Dtests.seed=98D53194946B5C85 -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=hi-IN -Dtests.timezone=Asia/Istanbul
```

Please let `./gradlew :plugins:ingest-attachment:check` run for a couple of hours, to make sure the test suite is stable, thank you.

**Comment by finnegancarroll** [comment]
Removed strict checksum validation for some additional files with locale dependent parsing. Ran for a couple hours and with all available locales in `Locale.getAvailableLocales()` to ensure no flaky cases remain.

**Review by dblock — APPROVED** [review]
This looks better than what we have, @reta any objections?

**Comment by reta** [comment]
> This looks better than what we have, @reta any objections?

It really does, no objections @dblock , just double checking no flakyness is going to be introduced

**Comment by dblock** [comment]
> > This looks better than what we have, @reta any objections?
> 
> It really does, no objections @dblock , just double checking no flakyness is going to be introduced

Thanks. All yours to merge.

**Pull request merged**